spec/controllers/patron_controller_spec.rb

```diff
@@ -40,6 +40,28 @@
       expect(response.body).to eq("{\"netid\":\"steve\",\"first_name\":\"Steven\",\"last_name\":\"Smith\",\"barcode\":\"00000000000000\",\"barcode_status\":1,\"barcode_status_date\":\"2013-10-17T16:11:29.000-05:00\",\"university_id\":\"000000000\",\"patron_group\":\"staff\",\"purge_date\":\"2016-10-31T23:00:06.000-05:00\",\"expire_date\":\"2017-10-31T23:00:06.000-05:00\",\"patron_id\":\"0000\",\"campus_authorized\":false}")
     end
 
+    it "allows authenticated users to access patron info and ldap data when desired" do
+      stub_patron('steve')
+      user = double('user')
+      allow(request.env['warden']).to receive(:authenticate!) { user }
+      allow(controller).to receive(:current_user) { user }
+      expect(Ldap).to receive(:find_by_netid).with('steve').and_return({ldap_data: "is here"})
+      get :patron_info, params: { patron_id: 'steve', ldap: true, format: :json }
+      expect(response).to have_http_status(200)
+      expect(response.body).to eq("{\"netid\":\"steve\",\"first_name\":\"Steven\",\"last_name\":\"Smith\",\"barcode\":\"00000000000000\",\"barcode_status\":1,\"barcode_status_date\":\"2013-10-17T16:11:29.000-05:00\",\"university_id\":\"000000000\",\"patron_group\":\"staff\",\"purge_date\":\"2016-10-31T23:00:06.000-05:00\",\"expire_date\":\"2017-10-31T23:00:06.000-05:00\",\"patron_id\":\"0000\",\"campus_authorized\":false,\"ldap\":{\"ldap_data\":\"is here\"}}")
+    end
+
+    it "allows authenticated users to access patron info and ldap data when desired" do
```

Comment:
This test seems to need a different title

Reply:
@hackmastera I updated it.